src/BottomNavigation/BottomNavigation.react.js

```diff
@@ -44,20 +42,67 @@ function getStyles(props, context) {
 */
 
 class BottomNavigation extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            moveAnimated: new Animated.Value(0 - this.context.uiTheme.bottomNavigation.container.height),
+        };
+    }
+
+    componentDidMount() {
+        Animated.timing(this.state.moveAnimated, {
+            toValue: 0,
+            duration: 225,
+            easing: Easing.bezier(0.0, 0.0, 0.2, 1),
+        }).start();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.shouldShow !== this.props.shouldShow && nextProps.shouldShow === true) {
+            this._show();
+        }
+
+        if (nextProps.shouldHide !== this.props.shouldHide && nextProps.shouldHide === true) {
+            this._hide();
+        }
+    }
+
+    _show = () => {
+        Animated.timing(this.state.moveAnimated, {
+            toValue: 0,
+            delay: 0,
+            duration: 225,
+            easing: Easing.bezier(0.0, 0.0, 0.2, 1),
+        }).start();
+    }
+
+    _hide = () => {
+        Animated.timing(this.state.moveAnimated, {
+            toValue: 0 - this.context.uiTheme.bottomNavigation.container.height,
+            duration: 195,
+            easing: Easing.bezier(0.4, 0.0, 0.6, 1),
+        }).start();
+    }
+
     render() {
         const { active, children } = this.props;
-        const styles = getStyles(this.props, this.context);
+        const local = {
+          container: {
+            bottom: this.state.moveAnimated,
+          },
+        };
```

Comment:
**First of all:**
`getStyles` should return complete styles. So, you can move the `local` object to get styles method and pass the `this.state` to `getStyles` method as a third param. As I did it [here](https://github.com/xotahal/react-native-material-ui/blob/7e8df965475b3e7af0b57dd1ab0bc57b368a2cf1/src/Button/Button.react.js#L51)

**But:**
IMHO it's better to use it directly in animated.view since it's instance of `Animated.Value`. Because it's kind of internal state of current position of view and if someone passes style prop to BottomNavigation, he could rewrite your animated value, because user's style has always [the biggest priority](https://github.com/xotahal/react-native-material-ui/blob/master/src/BottomNavigation/BottomNavigation.react.js#L37)

```
<Animated.View style={[styles.container, { bottom: this.state.moveAnimated }]>